deployment/deploy/deploy.cpp

```diff
@@ -1135,6 +1151,9 @@ IPropertyTree* getInstances(const IPropertyTree* pEnvRoot, const char* compName,
       const char* buildSet= pComponent->queryProp("@buildSet");
       const char* logDir = NULL;
 
+      if ((!strcmp(buildSet,"thor") || !strcmp(buildSet,"roxie")) && !pTopologyComponents.contains(name))
```

Comment:
So the block from 1121 - 1135 is designed to parse Software/Topology/* and return a list of unique names for RoxieCluster/ThorCluster processes actually found in toplology.  Here on line 1154 we are in the loop that is iterating through Software/*  (which skips over Topology and Directories.)  We are matching any process found under Software/ that is a thor or roxie process, with the unique name that we found in topology.  If it matches we output it as a valid component to be listed.